apps/payment/models.py

```diff
@@ -417,52 +427,81 @@ class Meta:
     """Amount in NOK"""
-    used_stripe = models.BooleanField(default=False)
-    """Was transaction paid for using Stripe"""
-    datetime = models.DateTimeField(auto_now=True)
-    """Transaction creation datetime. Automatically generated."""
-    status = models.CharField(
-        max_length=30,
-        null=False,
-        choices=status.PAYMENT_STATUS_CHOICES,
-        default=status.SUCCEEDED,
-    )
-    """ Status of a Stripe payment """
-    stripe_id = models.CharField(max_length=128, null=True, blank=True)
-    """ID from Stripe payment"""
-    payment_intent_secret = models.CharField(max_length=200, null=True, blank=True)
-    """ Stripe payment intent secret key for verifying pending transactions/intents """
+    source = models.CharField(max_length=64, choices=TransactionSource.ALL_CHOICES)
+    """ Origin of the transaction, such as purchases in Nibble or additions from Stripe """
+    datetime = models.DateTimeField(auto_now_add=True)
+
+    @property
+    def used_stripe(self):
+        return self.source == TransactionSource.STRIPE
 
     def get_timestamp(self):
         return self.datetime
 
     def get_subject(self):
-        return "[Kvittering] Saldoinnskudd på online.ntnu.no"
+        return f"[Kvittering] {self.get_description()}"
 
     def get_description(self):
-        return "saldoinnskudd på online.ntnu.no"
+        if self.source == TransactionSource.STRIPE:
+            return "Saldoinnskudd på online.ntnu.no"
+        elif self.source == TransactionSource.TRANSFER:
+            return "Overføring av saldo til annen bruker på online.ntnu.no"
+        elif self.source == TransactionSource.CASH:
+            return "Innskudd av kontanter på online.ntnu.no"
+        elif self.source == TransactionSource.SHOP:
+            return "Kjøp i Onlinekiosken"
 
     def get_items(self):
-        items = [{"name": "Påfyll av saldo", "price": self.amount, "quantity": 1}]
-        return items
+        if self.source == TransactionSource.STRIPE:
+            return [{"name": "Påfyll av saldo", "price": self.amount, "quantity": 1}]
+        elif self.source == TransactionSource.TRANSFER:
+            return [
+                {"name": "Overføring av saldo", "price": self.amount, "quantity": 1}
+            ]
+        elif self.source == TransactionSource.CASH:
+            return [
+                {"name": "Påfyll av kontanter", "price": self.amount, "quantity": 1}
+            ]
+        elif self.source == TransactionSource.SHOP:
+            if hasattr(self, "shop_order_line"):
+                return self.shop_order_line.get_order_descriptions()
+            else:
+                raise ValueError(
+                    f"Transaction for a shop purchase is not connected to an OrderLine"
+                )
 
     def get_from_mail(self):
         return settings.EMAIL_TRIKOM
 
     def get_to_mail(self):
-        return self.user.email
+        return self.user.primary_email
```

Comment:
We should be a tad bit careful with this one, but I have checked it now and it appears that `email` is a string either way

Reply:
The `primary_email` field was specifically created to avoid these problems, so that we wouldn't have to worry about that :smile: